### PR TITLE
ENH:  Add normalizing flow model.

### PR DESCRIPTION
**Pull request by ntustison**

### CODE REVIEW

**Comment by ntustison** [comment]
Hey @stnava, just an update on this.  Found a really [nice tf/keras example](https://github.com/aganse/flow_models) that uses a native tensor_flow_probability implementation.  I'm going to try training it on some 2-D brain slices as soon as the remaining DeepAtropos networks finish (probably sometime this week).  Apparently, [pyro](https://pyro.ai/examples/normalizing_flows_i.html) is the Pytorch analog which I'll look at as soon as I get this working.
